backend/src/api/forks/forks.handler.ts

```diff
@@ -34,53 +52,140 @@ export const postForkHandler: AuthorizedEventHandler = async (e) => {
       }
     })
   }
-  const templateToUse = await getTemplate(forkArgs.templateId)
-  await github.createFork(e.githubToken, forkArgs.name, templateToUse)
-  const forkInDb = await prisma.fork.create({
+  const template = await prisma.forkTemplate.findUnique({ where: { id: templateId } })
+  if (!template) {
+    return buildJsonResponse(404, { message: `Template not found with ID: ${templateId}` })
+  }
+
+  const existing = await prisma.fork.findFirst({ where: { userId: githubUser.id, templateId } })
+  if (existing) {
+    return buildJsonResponse(400, { message: `Fork already exists with template ID: ${templateId}` })
+  }
+
+  await github.createFork(e.githubToken, name, template)
+  const fork = await prisma.fork.create({
     data: {
-      appName: forkArgs.name,
+      appName: name,
       owner: githubUser.login,
       userId: githubUser.id,
-      templateId: templateToUse.id
+      state: ForkState.CREATED,
+      templateId: template.id
     }
   })
-  return buildJsonResponse(201, forkInDb)
+  return buildJsonResponse(201, toExtendedFork(fork))
 }
 
 export const putSecretsHandler: AuthorizedEventHandler = async (e) => {
-  const secretArgs: ForkSecretArgs = e.body as ForkSecretArgs
+  const secretArgs: ForkAwsSecretArgs = e.body as ForkAwsSecretArgs
   const forkId = Number(e.pathParameters!.id)
   const githubUser = await github.getUser(e.githubToken)
-  const fork: ExtendedFork = await prisma.fork.findFirst({
-    where: { userId: githubUser.id, id: forkId }
+  const fork = await prisma.fork.findFirst({
+    where: { userId: githubUser.id, id: forkId },
+    include: { template: true }
   })
+
   if (!fork) {
-    return buildJsonResponse(404, { message: 'No such repo' })
+    return buildJsonResponse(404, { message: `Fork not found with ID: ${forkId}` })
+  }
+
+  const secrets = createProviderSecrets(
+    fork.template.provider as ForkTemplateProvider,
+    fork.appName,
+    secretArgs
+  )
+
+  if (!secrets) {
+    return buildJsonResponse(400, { message: `Invalid secrets` })
   }
+
   await createSecrets(
-    {
-      awsAccessKey: secretArgs.awsAccessKey,
-      awsDefaultRegion: secretArgs.awsDefaultRegion,
-      awsSecretKey: secretArgs.awsSecretKey
-    },
     e.githubToken,
     githubUser.login,
-    fork.appName
+    fork.appName,
+    secrets
   )
-  return { statusCode: 201, body: '' }
+
+  await prisma.fork.update({
+    where: { id: forkId },
+    data: { secretsSet: true }
+  })
+
+  return buildJsonResponse(204)
 }
 
 export const getSecretsHandler: AuthorizedEventHandler = async (e) => {
```

Comment:
Probably not needed anymore because of `ExtendedFork.secretsSet`.

Reply:
Voisi pitää vielä mukana debuggailua varten. Poistaa sitten lopullisesta.